tmpl/wiki/weekly/2015-03-25.md

```diff
@@ -0,0 +1,122 @@
+
+We need to think about this carefully and it will involve more in-person
+discussion. It's a hairball as things are not necessarily in the right places 
+currently) and there's the risk of confusion as people add their own entropy
+sources for specific purposes — poor changes propagate through people's code
+bases. It's not just consumers of entropy but also providers of entropy that
+need to be addressed.  This leads to several structural issues and we need to
+figure out what to do.  An open question is how many people expect to write
+code that needs an entropy interface.
+
+Getting the entropy story figured out will likely delay the next point-release
+of Mirage (2.4.0) — if it isn't fixed.  DavidK will send an email to the
+mailing list to continue this discussion.
+*\[Edit: The thread is ["Update on entropy"][mail-entropy].\]*
+
+[mail-entropy]: http://lists.xenproject.org/archives/html/mirageos-devel/2015-03/msg00117.html
+
+
+#### Quality and test #### 
+
+There's work towards a virtual network interface, which can run TCP and iperf
+tests at the moment.  The goal is to be able to test the network stack but
+without needing an *actual* network (test between unikernels). At the moment,
+it's just been tested with two unikernels and we've reported issues with `Lwt`.
+If (and only if) you add a delay on write, then it works ok (this was
+discussed on the mailing list).  Will stick with two unikernels for now and
+later on, we can add support for more.  Also done some work on netfront and
+pcap and used them to do some simple tests for now.  In time, we should be
+able to write some meaningful tests 
+
+At the moment, not many people have had a chance to look at coveralls and
+bisect on Dave's repos — Mindy's had a look but ThomasG hasn't yet.
+
+
+#### Mirage.io status ####
+
+Last status was that the second Bytemark machine was being reinstalled to run
+a version of Xenserver (with xapi).  The idea is that this would then host the
+Mirage website. We need Anil present to discuss this so this item is moved to
+the next call.
+
+
+#### 3.0 Planning ####
+
+Following on from the [previous discussion][prev-rdmp] a month ago, it's
+becoming clear that there's an emphasis on improving quality of the libs.
+which means better tests.  Amir will summarise where things are with the
+thoughts around the 3.0 release and send to the list before the next time we
+discuss this.
+
+[prev-rdmp]: http://openmirage.org/wiki/weekly-2015-02-25#30Planning
+[roadmap]: https://github.com/mirage/mirage-www/wiki/Roadmap
+
+-- 
+
+#### AoB ####
+
+- **Irmin update**: ThomasG is still working on getting feedback upstreamed
+and improving the API. The big thing missing is having a garbage collector.
+Over the weekend ThomasL transferred all his items to the browser and changed
+to using IndexDB which does not have a quota limit. ThomasL will submit a bug
+report for others to look at. Noted that a JS backend for Irmin as separate
```

Comment:
Hmm... I might have misheard 'blog' as 'bug' during the call - and noted down the wrong thing.